rechnung/invoice.py

```diff
@@ -40,69 +42,55 @@ def fill_invoice_positions(positions, n_months, settings):
     invoice_total_vat = round(invoice_total_gross - invoice_total_net, 2)
 
-    return (
-        invoice_positions,
-        invoice_total_net,
-        invoice_total_vat,
-        invoice_total_gross,
-    )
-
+    return (invoice_items, invoice_total_net, invoice_total_vat, invoice_total_gross)
 
-def generate_invoice(
-    customer, positions, start_date, end_date, n_months, year, suffix, settings
-):
 
-    invoice_positions, net, vat, gross = fill_invoice_positions(
-        positions, n_months, settings
-    )
+def generate_invoice(settings, contract, year, month):
+    invoice_items, net, vat, gross = fill_invoice_items(settings, contract["items"])
 
     invoice_data = {}
-    invoice_data["address"] = customer["address"]
-    invoice_data["cid"] = customer["cid"]
-    invoice_data["date"] = datetime.datetime.now().strftime("%d. %B %Y")
-    invoice_data["id"] = "{}.{}.{}".format(customer["cid"], year, suffix)
-    invoice_data["period"] = "{} - {}".format(start_date, end_date)
+    invoice_data["address"] = contract.get("address", ["", "", ""])
+    invoice_data["cid"] = contract["cid"]
+    invoice_data["date"] = datetime.datetime.now().strftime(
+        settings.delivery_date_format
+    )
+    invoice_data["id"] = f"{contract['cid']}.{year}.{month:02}"
+    invoice_data["period"] = "{}-{}".format(year, month)
     invoice_data["total_gross"] = gross
     invoice_data["total_net"] = net
     invoice_data["total_vat"] = vat
     invoice_data["vat"] = settings.vat
-
-    if "email" not in customer.keys():
-        invoice_data["email"] = None
-    else:
-        invoice_data["email"] = customer["email"]
-
-    return (invoice_data, invoice_positions)
+    invoice_data["email"] = contract["email"]
+    invoice_data["items"] = invoice_items
+    return invoice_data
 
 
 def iterate_invoices(invoices_dir):
     """
-    Generator which iterates over all customer directories and
-    included invoice yamls, yields customer_invoice_dir and filename.
+    Generator which iterates over all contract directories and
+    included invoice yamls, yields contract_invoice_dir and filename.
     """
     for d in os.listdir(invoices_dir):
-        customer_invoice_dir = os.path.join(invoices_dir, d)
-        if os.path.isdir(customer_invoice_dir):
-            for filename in os.listdir(customer_invoice_dir):
+        contract_invoice_dir = os.path.join(invoices_dir, d)
+        if os.path.isdir(contract_invoice_dir):
+            for filename in os.listdir(contract_invoice_dir):
 
                 if not filename.endswith(".yaml"):
                     continue
 
-                yield customer_invoice_dir, filename
+                yield contract_invoice_dir, filename
 
 
-def render_pdf_invoices(directory, template, settings):
-
-    logo_path = os.path.join(settings.assets_dir, "logo.svg")
+def render_invoices(settings):
+    template = get_template(settings.invoice_template_file)
+    logo_path = Path(settings.assets_dir / "logo.svg")
```

Comment:
We should define a setting "logo" because it's hard coded right now, and in _contract.py_ it's **logo.png** vs. **logo.svg** here. m)